### PR TITLE
Initial commit for a devon4ng tutorial

### PR DESCRIPTION
**Pull request by suprishi**

### CODE REVIEW

File: devon4ng-mat-layout/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+restoreDevonfwIde(["java","mvn", "npm", "ng"])
+--
+====
+Before creating a devon4ng application, you first have to install the devonfw ide. You will find more information about devonfw on https://devonfw.com/website/pages/welcome/welcome.html.
+Once you have setup devonfw ide, you are ready to create your first devon4ng project.
+[step]
+--
+createDevon4ngProject("devon4ng-mat-layout", "")
+--
+Enter
```
Comment (GuentherJulian) [inline devon4ng-mat-layout/index.asciidoc:17]: This syntax is not allowed. Everything below '--' will be rendered as plain text until the next '====' appears. So the step runClientNg will also be displayed as text

File: devon4ng-mat-layout/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+`y` {{execute T1}} 
+
+when it asks whether it would like to add Angular routing and select `SCSS` when it asks for the stylesheet format.
```
Comment (GuentherJulian) [inline devon4ng-mat-layout/index.asciidoc:21]: You can also pass parameters to createDevon4ngProject as described here: https://github.com/devonfw-tutorials/tutorials/wiki/Functions
So you can simply write createDevon4ngProject("devon4ng-mat-layout", ["--style=scss"])

File: devon4ng-mat-layout/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+====
+Before creating a devon4ng application, you first have to install the devonfw ide. You will find more information about devonfw on https://devonfw.com/website/pages/welcome/welcome.html.
+Once you have setup devonfw ide, you are ready to create your first devon4ng project.
+[step]
+--
+createDevon4ngProject("devon4ng-mat-layout", "")
+--
+Enter
+
+`y` {{execute T1}}
```
Comment (GuentherJulian) [inline devon4ng-mat-layout/index.asciidoc:19]: We set NG_CLI_ANALYTICS=CI as environment variable. So I think this is not needed

**Review by GuentherJulian — CHANGES_REQUESTED** [review]

**Comment by suprishi** [comment]
@GuentherJulian is there any mistake with the way I have used the `changeFile` function. I'm always getting `directory does not exist` error.

**Comment by GuentherJulian** [comment]
> @GuentherJulian is there any mistake with the way I have used the `changeFile` function. I'm always getting `directory does not exist` error.

I think the problem is the npmInstall step. You have to pass the directory, in which the command is to be executed, as the first parameter.
`npmInstall("devon4ng-mat-layout", {"name": "@angular/material @angular/cdk @angular/animations", "args": ["--save"]})`

File: .github/workflows/buildMain.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: CI
+
```
Comment (GuentherJulian) [inline .github/workflows/buildMain.yml:2]: Why are these two workflow files part of this PR? Maybe you have to update your fork and create a new PR.

Comment (suprishi) [reply .github/workflows/buildMain.yml:2]: No clue about when these files appeared in my PR. I will update my fork. Shall I delete this PR and create a new one?

Comment (GuentherJulian) [reply .github/workflows/buildMain.yml:2]: Yes, I think this should solve the problem

**Comment by suprishi** [comment]
Closing this one as updated one is under another [PR](https://github.com/devonfw-tutorials/tutorials/pull/77)

**Pull request closed without merging**